### PR TITLE
[bitnami/redis] fixed the bug related to issue #7283

### LINKED ISSUE #7283: [bitnami/redis] endless crash loop if sentinel mode is active
**Which chart**:
bitnami/redis 13.8.11

**Describe the bug**
Redis pods are stuck in a crash loop without possible way to receover

redis cluster contains 3 replicas and all are running without PVC

**To Reproduce**
difficult, that happens regularly but didn't find a way to reproduce

it happens if cluster nodes are replaced or pods are evicted

just have a possible way to reproduce:

1. let one replica come up
2. wait for the pod creation of the second one
3. before 2nd node pod was started kill the first one
4. endless crash loop

**Expected behavior**
cluster is recovering even if no replica is running

**Version of Helm and Kubernetes**:
helm: v3.6.3
kubernetes: 1.21.2

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.6.3", GitCommit:"d506314abfb5d21419df8c7e7e68012379db2354", GitTreeState:"dirty", GoVersion:"go1.16.6"}
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.1", GitCommit:"632ed300f2c34f6d6d15ca4cef3d3c7073412212", GitTreeState:"clean", BuildDate:"2021-08-19T15:38:26Z", GoVersion:"go1.16.6", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.2", GitCommit:"7ea22988e84a8e0701cb7e4b21b44d2634f8205d", GitTreeState:"clean", BuildDate:"2021-07-15T22:09:43Z", GoVersion:"go1.16.5", Compiler:"gc", Platform:"linux/amd64"}
```

**Additional context**
`svc/redis` doesn't have active endpoints because all pods are crashing and not getting ready, possible race condition?

but they are trying to connect to `svc/redis`, shouldn't they try to connect to `svc/redis-headless`?
`svc/redis` will never work if they are stuck in crash loop as no redis pod will every get ready

```
redis Could not connect to Redis at redis.xxxxxxx.svc.cluster.local:26379: Connection timed out
redis Could not connect to Redis at -p:6379: Name or service not known
sentinel Could not connect to Redis at redis.xxxxxxx.svc.cluster.local:26379: Connection timed out
sentinel Could not connect to Redis at -p:6379: Name or service not known
```

### PR DESCRIPTION
**Pull request by AminSojoudi**
**Description of the change**

**Benefits**

Redis sentinels are now much more resilient to chaos and can recover themselves in most situations.

**Possible drawbacks**

It may possible that this change creates multiple masters for a very limited time in the process of recovery.

**Applicable issues**

  - fixes #7283

**Additional information**

**Checklist** 

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])

### CODE REVIEW

File: bitnami/redis/templates/scripts-configmap.yaml [hunk 1]
```diff
@@ -318,7 +318,7 @@ data:
     {{- end }}
     printf "\nsentinel myid %s" "$(host_id "$HOSTNAME")" >> /opt/bitnami/redis-sentinel/etc/sentinel.conf
 
-    if [[ -z "$(getent ahosts "$HEADLESS_SERVICE" | grep -v "^${myip}")" ]]; then
+    if [[  ! $(get_sentinel_master_info) ]]; then
```
Comment (javsalgar) [inline bitnami/redis/templates/scripts-configmap.yaml:321]: I don't think you need the `[[` and `]]` in this case. Could you try removing it?

**Review by javsalgar — CHANGES_REQUESTED** [review]
Thank you so much! Changes requested

File: bitnami/redis/templates/scripts-configmap.yaml (hunk 1 above)
Comment (AminSojoudi) [reply bitnami/redis/templates/scripts-configmap.yaml:321]: sure, I will do that now

**Comment by juan131** [comment]
@AminSojoudi the CI tests are complaining about the chart version not being updated. It's very likely that you need to rebase your branch from master and force-push the changes to update it.

**Comment by juan131** [comment]
Hi @AminSojoudi, the chart version is still not updated in the **Chart.yaml**

**Review by juan131 — APPROVED** [review]
LGTM

**Pull request merged**